### DISCUSSION TITLE
updating config for a device requiring a rebuilt kernel

### DISCUSSION THREAD

**Discussion by phartshorn** [onitake/gsl-firmware#181] [General]
Just a quick discussion point. Who thought having to recompile the kernel to change configuration for the driver (modify quirks) was a good thing? I've recently set up android on an old tablet that uses these drivers. I'm installing from an iso from android-x86 and I was surprised that there was a change from allowing quick changes to the quirks (via fw_tool) to requiring a kernel recompile to do the same.

Would it be possible to allow the quirks to be set either via a configuration file or the kernel boot params, or even supporting the old quirks being included in the firmware? For having to recompile the kernel to support a new configuration is a major step backwards.

I'm just fortunate that the "deprecated, yet more functional" gslx680 kernel module was included in the build.

**Comment by onitake** [comment]
Hi @phartshorn , there was some recent discussion about this on linux-input: https://patchwork.kernel.org/project/linux-input/list/?series=499519

Unfortunately, no action was taken in the end, because there were concerns about the safety and robustness of the parsing code.
I think it would be best to pick up from there.

**Comment by onitake** [comment]
@jwrdegoede Are you still working on this patch set?
